py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py

```diff
@@ -473,15 +474,47 @@ def run(
         Args:
             strict_type_constraints: Usually we should set it to False unless we want to control the precision of certain layer for numeric reasons.
             algorithm_selector: set up algorithm selection for certain layer
+            tactic_sources: set up tactic sources for certain layer
         Return:
             TRTInterpreterResult
         """
+        if (
+            self.compilation_settings.custom_engine_cache is not None
+        ):  # custom_engine_cache could be None if this function is called from convert_exported_program_to_serialized_trt_engine etc.
+            if (
+                self.compilation_settings.cache_built_engines
+                or self.compilation_settings.reuse_cached_engines
+            ):
+                engine_cache = self.compilation_settings.custom_engine_cache
+                hash_val = engine_cache.get_hash(self.module)
+
+            if self.compilation_settings.reuse_cached_engines:
+                # query the cached TRT engine
+                blob = engine_cache.load(hash_val)
+                if blob is not None:  # hit the cache
+                    serialized_engine, input_names, output_names, weight_name_map = (
+                        engine_cache.unpack(blob)
+                    )
+                    self._input_names = input_names
+                    self._output_names = output_names
+                    self.weight_name_map = weight_name_map
+                    _LOGGER.info(
+                        "Found the cached engine that corresponds to this graph. It is directly loaded."
+                    )
+                    return TRTInterpreterResult(
```

Comment:
Where does refit happen?

Reply:
Are we always going to make weight stripped engines now or something?

Reply:
Adrian said the **fast refit** would be called even without calling engine caching, so that's why it's after `interpreter.run()`: https://github.com/pytorch/TensorRT/pull/2995/files#diff-5d0cff736de42c72a7c404440a15e371fac0984794a5de7e651cc2a32566856dR111-R129

I just realized that it's incorrect to put refitting code after `interpreter.run()` because `interpreter.run()` could be called in other places. In these scenarios, the engine weights will be 0 if use engine caching. I plan to move it into `interpreter.run()`.